base_layer/core/src/mining/miner.rs

```diff
@@ -187,7 +189,8 @@ impl Miner {
 
         pin_mut!(block_event);
         pin_mut!(state_event);
-        let mut start_mining = false;
+        // Start mining immediately in case we're the only node on the network and we never receive a BlockSync event
+        let mut start_mining = true;
```

Comment:
I think this should still wait for the BlocksSynchronized event

Reply:
Bit of a trade-off here for sure.